service-csw/src/main/java/fi/nls/oskari/search/channel/MetadataCatalogueQueryHelper.java

```diff
@@ -293,9 +295,14 @@ private String parseWKTPolygon(final String searchCriterion) {
             fc = fjs.readFeatureCollection(new ByteArrayInputStream(
                     searchCriterion.getBytes("utf-8")));
             ReferencedEnvelope env = fc.getBounds();
+
+            CoordinateReferenceSystem sourceCRS = env.getCoordinateReferenceSystem();
+            if (sourceCRS == null) {
+                sourceCRS = CRS.decode("EPSG:4326", true); // default CRS in GeoJSON
```

Comment:
Since TARGET_SRS is also always "EPSG:4326", no CRS.decode nor ProjectionHelper.transformPoint methods are required. That said I think that doing it this way makes the code more readable and clear.

Reply:
Yes, you're right. But if someone changes TARGET_SRS constant to something else, the above will still work without changes.